io/io/src/TFile.cxx

```diff
@@ -4030,6 +4036,22 @@ TFile *TFile::Open(const char *url, Option_t *options, const char *ftitle,
       return f;
    }
 
+#ifdef R__UNIX
+   // If URL is a file on an EOS FUSE mount, attempt redirection to XRootD protocol.
+   if (gEnv->GetValue("TFile.CrossProtocolRedirects", 1) == 1) {
+      TUrl fileurl(url, /* default is file */ kTRUE);
+      if (strcmp(fileurl.GetProtocol(), "file") == 0) {
+         ssize_t len = getxattr(fileurl.GetFile(), "eos.url.xroot", nullptr, 0);
+         if (len > 0) {
+            std::string xurl(len, 0);
+            if (getxattr(fileurl.GetFile(), "eos.url.xroot", &xurl[0], len) == len)
+               if ((f = TFile::Open(xurl.c_str(), options, ftitle, compress, netopt)))
```

Comment:
I didn't see a zombie being returned, even when I tested without proper credentials, but I added a check just in case. When you can read via `/eos` but not via `root://`, a lot of error messages about credentials appear in the output:
```
Error in <TNetXNGFile::Open>: [ERROR] Server responded with an error: [3010] Unable to give access - user access restricted - unauthorized identity used ; Permission denied
```
However, in the end you get a `nullptr` from the attempted `TFile::Open` and fallback to opening the file via FUSE.